staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion_test.go

```diff
@@ -0,0 +1,284 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apiextensionsfeatures "k8s.io/apiextensions-apiserver/pkg/features"
+	"k8s.io/apiextensions-apiserver/test/integration/converter"
+	"k8s.io/apiextensions-apiserver/test/integration/fixtures"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/util/retry"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConvert(t *testing.T) {
+	// set up webhook server
+	roots := x509.NewCertPool()
+	if !roots.AppendCertsFromPEM(localhostCert) {
+		t.Fatal("Failed to append Cert from PEM")
+	}
+	cert, err := tls.X509KeyPair(localhostCert, localhostKey)
+	if err != nil {
+		t.Fatalf("Failed to build cert with error: %+v", err)
+	}
+	webhookMux := http.NewServeMux()
+	webhookMux.Handle("/crdconvert", http.HandlerFunc(converter.Serve))
+	webhookServer := httptest.NewUnstartedServer(webhookMux)
+	webhookServer.TLS = &tls.Config{
+		RootCAs:      roots,
+		Certificates: []tls.Certificate{cert},
+	}
+	webhookServer.StartTLS()
+	defer webhookServer.Close()
+
+	// start apiserver
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, apiextensionsfeatures.CustomResourceWebhookConversion, true)()
+	tearDown, apiExtensionClient, dynamicClient, err := fixtures.StartDefaultServerWithClients(t)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tearDown()
+
+	// create test CRD
+	url := webhookServer.URL + "/crdconvert"
+	crd := multiVersionConversionFixture.DeepCopy()
+	crd.Spec.Conversion = &apiextensionsv1beta1.CustomResourceConversion{
+		Strategy: apiextensionsv1beta1.WebhookConverter,
+		WebhookClientConfig: &apiextensionsv1beta1.WebhookClientConfig{
+			URL:      &url,
+			CABundle: localhostCert,
+		},
+	}
+	crd = CreateCRDAndWaitForDiscovery(t, apiExtensionClient, false, crd)
+
+	versionedClients := map[string]dynamic.ResourceInterface{}
+	for _, v := range crd.Spec.Versions {
+		versionedClients[v.Name] = dynamicClient.Resource(schema.GroupVersionResource{Group: crd.Spec.Group, Version: v.Name, Resource: crd.Spec.Names.Plural})
+	}
+	testCustomResourceConversionWebhook(t, crd, versionedClients)
+	testCRListConversion(t, crd, apiExtensionClient, versionedClients)
```

Comment:
nit: clean up the cluster scoped CRD and CRs? 

we are using a unique etcd-prefix so maybe it's fine

Reply:
we can do all that in a follow. @jpbetz has his toughts about a good test structure probably. These are only the tests from e2e to make them available here and not to regress. Would leave it for the moment.

Reply:
rebased on @jpbetz's test framework.